internal/provider/resource_aci_pim_route_map_entry.go

```diff
@@ -0,0 +1,595 @@
+			},
+			"description": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+				MarkdownDescription: `The description of the Pim Route Map Entry object.`,
+			},
+			"grp": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+				MarkdownDescription: `Multicast group ip/prefix.`,
+			},
+			"name": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+				MarkdownDescription: `The name of the Pim Route Map Entry object.`,
+			},
+			"name_alias": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+				MarkdownDescription: `The name alias of the Pim Route Map Entry object.`,
+			},
+			"order": schema.StringAttribute{
+				Required: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+					stringplanmodifier.RequiresReplace(),
+				},
+				MarkdownDescription: `PIM route map entry order.`,
+			},
+			"rp": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+				MarkdownDescription: `Multicast RP Ip.`,
+			},
+			"src": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+				MarkdownDescription: `Multicast Source Ip.`,
+			},
+			"annotations": schema.SetNestedAttribute{
+				MarkdownDescription: ``,
+				Optional:            true,
+				Computed:            true,
+				PlanModifiers: []planmodifier.Set{
+					setplanmodifier.UseStateForUnknown(),
+				},
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"key": schema.StringAttribute{
+							Required: true,
+							PlanModifiers: []planmodifier.String{
+								stringplanmodifier.UseStateForUnknown(),
+							},
+							MarkdownDescription: `The key or password used to uniquely identify this configuration object.`,
+						},
+						"value": schema.StringAttribute{
+							Required: true,
+							PlanModifiers: []planmodifier.String{
+								stringplanmodifier.UseStateForUnknown(),
+							},
+							MarkdownDescription: `The value of the property.`,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func (r *PimRouteMapEntryResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	tflog.Trace(ctx, "start configure of resource: aci_pim_route_map_entry")
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
```

Comment:
Why not display an error message? Is this caught elsewhere?

Reply:
I have not look into this into detail and also do not know how we would get into a situation where this is nil, but I assume this is panic prevention is captured by the framework logic. 

In the framework examples you see that they use this https://github.com/hashicorp/terraform-provider-scaffolding-framework/blob/main/internal/provider/example_resource.go#L73-L76

If there is a scenario where it would be nil and this nil check is not there there would be a panic raise. Which we would like to avoid.

Reply:
will add diagnostic for certainty

Reply:
added diagnostics in commit https://github.com/CiscoDevNet/terraform-provider-aci/pull/1113/commits/eb29147a2390d7dcf44bb16856687dd426b7e6a0

also applied same change on data source side

Reply:
reverted the change since test are failing, appears that there are occasions where defined configure function in framework is called with empty data, so none of the resource will be able to execute